### ISSUE TITLE
Return `subTags` as part of the `getTag` function.

### ISSUE THREAD

**Issue by darthnithin** [FujoWebDev/AO3.js#69]
We can trivially get Parent tags using the getTag function, but that function doesn't seem to expose child tags, despite that being available on AO3? Metatags also seem to be inaccessable.

**Comment by darthnithin** [comment]
This makes it not possible to traverse tags.

**Comment by essential-randomness** [comment]
Yup, that's not currently available! I've asked in our fandom coders Discord whether anyone wants to take a stab at it, so let me give some details on what it would need to be added:

1. Implementation of the subtag parsing would go in the [/src/tags/page-getters.ts](https://github.com/FujoWebDev/AO3.js/blob/main/src/tags/page-getters.ts) file
1. The method should then be used as part of [the `getTag` function](https://github.com/FujoWebDev/AO3.js/blob/main/src/tags/index.ts#L25)
2. And here is the [Tag type](https://github.com/FujoWebDev/AO3.js/blob/main/types/entities.ts#L8) that would need to be updated.

The [Fluff tag](https://archiveofourown.org/tags/Fluff) seems like a good tag to write tests for this (though we should have tests for one page without subtags). Tests would go in [this file](https://github.com/FujoWebDev/AO3.js/blob/main/tests/tags.test.ts).

The one difficulty I see is that some subtags also have subtags and those are also displayed on the parent tag page.

I see two solutions:
1. We only return first-level subtags
2. We always return subtags as "`subTags: { tagName: string; subTags: Array<string> }`" with array being empty by default. 

I have a slight preference for 2 since people would need to query every single subtag to know if subtags existed without it, though i don't know if there's cases where AO3 doesn't surface subtags there even if they exist. It complicates code a little bit compared to having `subTags: Array<string>` as a type (solution 1), but it's both more future proof and not that huge of a complication.


Thank you for reporting this as a need and using our library :D hopefully we can get a taker for the issue.

**Comment by rhyses-pieces** [comment]
Hey there, I'd like to take a stab at this! I'm working on getting the repo set up and figuring out the implementation.

**Comment by essential-randomness** [comment]
> Hey there, I'd like to take a stab at this! I'm working on getting the repo set up and figuring out the implementation.

Seems like someone else in our Discord might already be working on it (sorry about not assigning the issue, just woke up). However, if you've gotten the thing setup we could also use the 'Tags with the same meaning' section, which should have very similar implementation! 

![image](https://github.com/user-attachments/assets/35080e1c-58c6-4e6f-98bc-04e4a342a494)

(Please disregard if you're the same person who said they were doing this in fandom coders!)

**Comment by essential-randomness** [comment]
Nevermind, rhyses is indeed the same person as our fandom coders' tribute 👯

**Comment by darthnithin** [comment]
How does one join the discord server? Perhaps add the link to Contributing.md? Or did I just miss something?

**Comment by darthnithin** [comment]
https://github.com/FujoWebDev/AO3.js/pull/75 solves getting Subtags, but we still aren't getting Child tags.

**Comment by essential-randomness** [comment]
> How does one join the discord server? Perhaps add the link to Contributing.md? Or did I just miss something?

We don't publicize the link prominently since we're still working on scaling moderation, but if you [submit an application here](https://www.fancoders.com/#join) and let us know who you are we'll send an invite shortly :) 

I saw some comments on the PR so i'll go continue the rest of the convo there!

**Comment by darthnithin** [comment]
> #75 solves getting Subtags, but we still aren't getting Child tags.

#83

**Comment by essential-randomness** [comment]
Released [@bobaboard/ao3.js@0.18.0](https://www.npmjs.com/package/@bobaboard/ao3.js) (new minor) which includes this feature!

**Comment by darthnithin** [comment]
I know the scope of this issue has drifted a little, but the original problem of not being able to get child tags still exists. The subtag PR is great though!

**Comment by essential-randomness** [comment]
Now fully fixed with #83 (thank you!)

Published it with Released [@bobaboard/ao3.js@0.20.0](https://www.npmjs.com/package/@bobaboard/ao3.js) (new minor*) which includes this feature!

*I accidentally republished the same as @0.18 in @0.19, so we're skipping one!
